packages/components/src/components/step/step.ts

```diff
@@ -0,0 +1,176 @@
+  @property({ reflect: true }) state: 'waiting' | 'inProgress' | 'finished' = 'waiting';
+
+  /** Removes the tail from the step. */
+  @property({ reflect: true, type: Boolean, attribute: 'no-tail' }) noTail = false;
+
+  /** The step's label overwriting the `label` slot. Use the `label` slot for complex label content. */
+  @property() label = '';
+
+  /** The step's description overwriting the `description` slot. Use the `description` slot for complex description content. */
+  @property() description = '';
+
+  /** The step's number in a step-group */
+  @property({ type: Number, reflect: true, attribute: 'step-index' }) StepIndex = 1;
+
+  /** When set, the underlying button will be rendered as an `<a>` with this `href` instead of a `<button>`. */
+  @property() href = '';
+
+  connectedCallback() {
+    super.connectedCallback();
+  }
+
+  private isLink() {
+    return this.href ? true : false;
+  }
+
+  render() {
+    const isLink = this.isLink();
+    const tag = isLink ? literal`a` : literal`button`;
+
+    /* eslint-disable lit/no-invalid-html */
+    /* eslint-disable lit/binding-positions */
+    return html`
+      <div
+        part="base"
+        class=${cx(
+          'flex overflow-hidden pt-1',
+          this.orientation === 'horizontal' ? 'flex-col w-full' : 'flex-row gap-8 items-stretch h-full w-min',
+          this.state === 'waiting' ? '!text-neutral-500 ' : ''
+        )}
+      >
+        <div
+          part="circle-and-tail-container"
+          class=${cx(
+            'flex shrink-0 gap-4',
+
+            this.orientation === 'horizontal' ? 'flex-row' : 'flex-col items-stretch',
+            this.orientation === 'horizontal'
+              ? this.size === 'lg'
+                ? 'translateLg'
+                : 'translateSm'
+              : this.size === 'lg'
+                ? 'mt-1'
+                : 'mt-3'
+          )}
+        >
+          
+          <${tag}
+            part="circle"
+            ?disabled=${this.state !== 'finished'}
+            tabindex=${this.state === 'finished' ? '0' : '-1'}
+            href=${ifDefined(isLink ? this.href : undefined)}
+            aria-label=${this.state === 'finished' && 'Step completed'}
+            class=${cx(
+              'border rounded-full aspect-square circle flex items-center justify-center shrink-0 font-bold select-none',
+              this.state === 'finished' ? 'focus-visible:focus-outline' : 'focus-visible:outline-none',
+              this.size === 'lg' ? 'w-12' : 'w-8',
+              this.state === 'waiting' && 'border-neutral-500',
+              this.state === 'finished' && 'border-primary hover:bg-primary-100 hover:border-primary-500',
+              this.state === 'inProgress' && 'bg-accent border-none text-white'
+            )}
+          >
+            ${
+              this.state === 'finished'
+                ? html` <sd-icon name="confirm" library="system" color="primary"></sd-icon>`
+                : this.StepIndex
```

Comment:
Please make this a slot as well. People should be able to override this icons or even set their own content or whatever. Maybe you'll need two slots, `<slot name="complete-icon">`, `<slot name="index">`

Reply:
See a real life example from the website where the index should be something different:

![CleanShot 2024-05-03 at 16 26 31@2x](https://github.com/solid-design-system/solid/assets/26542182/0e506a35-9272-4859-bafb-0c9f0594de5f)

Reply:
![CleanShot 2024-05-03 at 16 29 27@2x](https://github.com/solid-design-system/solid/assets/26542182/5a9aca0f-a0e8-4cf9-8e9d-d0919abf9d8e)